app/components/StudioEditor.vue.ts

```diff
@@ -229,7 +245,7 @@ export default class StudioEditor extends Vue {
       });
     });
 
-    this.scene.getItem(source.sceneItemId).setPositionAndCrop(rect.x, rect.y, rect.crop);
+    this.scene.getItem(source.sceneItemId).setSettings({ x: rect.x, y: rect.y, crop: rect.crop });
```

Comment:
When I started working on Undo/Redo, I had a branch that rolled up a lot of these methods into a single function called `setTransform`, that took an `ITransform` (or maybe a `Partial<ITransform>`).

An `ITransform` looked like this:
```typescript
interface ITransform {
  position: IVec2;
  scale: IVec2;
  crop: ICrop;
  rotation: number;
}
```

I honestly thought I merged this branch, but I guess I didn't.  I think this was a good system because it grouped together some very logical things into a single object describing the position/shape of the item.  The undo/redo system was going to store any change to this object as a "Transform" event and store the before/after state of this object.

What do you think about this approach?  Maybe since we're in here we should make that change now?  Let me see if I can find that branch anywhere...

Reply:
My main goal in this refactoring was to rid of multiples `sceneItemChanged` events that makes a lot of spam in our API events when our code do something like:
```
item.setCrop(crop);
item.setPosition(position);
item.setVisible(visible);
// ^ 3 sceneItemChanged will be generated here!
```

So I created an interface 
```
ISceneItemSettings {
  x: number;
  y: number;
  scaleX: number;
  scaleY: number;
  visible: boolean;
  crop: ICrop;
  locked: boolean;
  rotation: number;
}
```
It's almost the same as ITransform but it also contains `locked` and `visible` values that I think the undo system can also store. I think I can change ISceneItemSettings to something like:
```
ISceneItemSettings {
  transform: ITransform,
  visible: boolean;
  locked: boolean;
}
```

I think it's a good idea to have a Transform class instead of `Rectalnge`  and `ScalableRectangle` that also can provide such things as `width` and `height`. For example, in Unity3d each object has `transform` property with `position` and `rotation` fields that manage the all Math with transforms.

Reply:
I think it's important that `width` and `height` are kept out of the Transform.  Width and height are properties of the underlying source, whereas a Transform can be applied to each scene item separately.